rpc2/codec.go

```diff
@@ -1,5 +1,55 @@
 package rpc2
 
+import (
+	"bytes"
+	"github.com/ugorji/go/codec"
+	"io/ioutil"
+)
+
 type Decoder interface {
 	Decode(interface{}) error
 }
+
+type Encoder interface {
+	Encode(interface{}) error
+}
+
+type ByteEncoder interface {
+	EncodeToBytes(interface{}) ([]byte, error)
+}
+
+type FramedMsgpackEncoder struct {
+	handle codec.Handle
+}
+
+func NewFramedMsgpackEncoder(mh *codec.MsgpackHandle) *FramedMsgpackEncoder {
+	if mh == nil {
+		mh = &codec.MsgpackHandle{WriteExt: true}
+	}
+	return &FramedMsgpackEncoder{
+		handle: mh,
+	}
+}
+
+func (e *FramedMsgpackEncoder) encodeToBytes(i interface{}) (v []byte, err error) {
+	buf := new(bytes.Buffer)
+	enc := codec.NewEncoder(buf, e.handle)
```

Comment:
I was attempting to maintain parity of functionality with before the refactor. Also, to use `NewEncoderBytes`, we'd have to know the length of `i`, which we can only get by using reflection. I'd rather avoid bringing in `reflect` magic when possible.